tests/vm/buttonservo.spec.ts

```diff
@@ -37,45 +39,78 @@ suite("button servo", () => {
             const runner = new VMProgramRunner(roleMgr, program)
             await runner.startAsync()
 
-            await testBody(tester, button, servo)
+            const servoReg = new RegisterTester(
+                servo.service.register(ServoReg.Angle)
+            )
+            await testBody(tester, button.server, servo.server, servoReg)
         })
     }
 
     test(
         "inverts when pressed",
-        makeVmTest(async (tester, button, servo) => {
-            servo.server.angle.setValues([50])
+        makeVmTest(async (tester, button, servo, servoReg) => {
+            servo.angle.setValues([50])
 
-            button.server.down()
-            await tester.waitForDelay(100)
-            button.server.up()
-            assert(servo.server.angle.values()[0] == -50.0)
+            await tester.assertWith(
+                servoReg
+                    .onUpdate({
+                        triggerRange: [-50, -50],
+                    })
+                    .hold(),
```

Comment:
i have a hard time grasping what hold is supposed to mean. can we use a more explicit (potentially longer) API name?

Reply:
Yeah, fair point. The idea is to have it trigger when the value is within some range, but it also supports triggering on any update (which isn't used).

It probably makes more sense to nix the second and name it so it's specific to the first - perhaps `onValue` - and require the triggerRange as the first argument while moving other conditions to a secondary options argument.